src/rust/src/decoder/tv_screen.rs

```diff
@@ -358,6 +360,140 @@ impl dtvcc_tv_screen {
         Ok(())
     }
 
+    fn count_captions_lines_scc(&self) -> usize {
+        (0..CCX_DTVCC_SCREENGRID_ROWS)
+            .filter(|&row_index| !self.is_row_empty(row_index as usize))
+            .count()
+    }
+
+    /// Write captions in SCC format
+    pub fn write_scc(&self, writer: &mut Writer) -> Result<(), String> {
+        fn adjust_odd_parity(value: u8) -> u8 {
+            let mut ones = 0;
+            for i in 0..=7 {
+                if value & (1 << i) != 0 {
+                    ones += 1;
+                }
+            }
+            if ones % 2 == 0 {
+                0b10000000 | value
+            } else {
+                value
+            }
+        }
+        fn add_needed_scc_labels(buf: &mut String, subtitle_count: usize, count: usize) {
```

Comment:
```suggestion
        fn add_needed_scc_labels(buf: &mut String, total_subtitle_count: usize, subtitles_displayed: usize) {
```
Could you please add some documentation about this function, particularly mentioning in which cases it choose row 15 vs row 14 vs row 13

Reply:
Yeah sure

Reply:
Done @PunitLodha